picard/util/__init__.py

```diff
@@ -926,13 +926,25 @@ def wrap_count(p):
     return fmt.format(title=default_title, count=count + 1)
 
 
-def get_base_title_with_suffix(title, suffix):
+def get_base_title_with_suffix(title, suffix, fmt=None):
     """Extract the base portion of a title,
        removing the suffix and number portion from the end.
     """
+    if fmt is None:
+        fmt = _('{title} ({count})')
+
+    parts = fmt.split('{title}')
+
+    def wrap_count(p):
+        if '{count}' in p:
+            return '(?:' + re.escape(p).replace('\\{count\\}', '\\d*') + ')?'
+        else:
+            return p
+
     escaped_suffix = re.escape(suffix)
-    re_text = "^(.*)\\s+" + escaped_suffix + "(\\s+\\(\\d*\\))?$"
-    match_obj = re.fullmatch(re_text, title)
+    joiner = '(.*?)(?:\\s*' + escaped_suffix + ')?'
+    regstr = '^' + joiner.join([wrap_count(p) for p in parts]) + '$'
+    match_obj = re.fullmatch(regstr, title)
```

Comment:
Alternative suggestion, which I think is a bit more clear to understand:

```python
reg_title = r'(?P<title>.*?)(?:\s*' + escaped_suffix + ')?'
reg_count = r'(?:\d+)?'
regstr = '^' + re.escape(fmt)
    .replace(r'\{title\}', reg_title)
    .replace(r'\{count\}', reg_count)
    .replace(' ', r'\s+') + '$'
match_obj = re.fullmatch(regstr, title)
return match_obj['title'] if match_obj else title
```

Reply:
I tried out your suggested code, but it fails most of the tests because it doesn't take into account the brackets around `{count}` in the format string.  I'll play with it a bit and see if I can work around that.

Reply:
I think this latest commit incorporates your suggestion to improve readability and clarifies what's happening.

Reply:
We can reduce code redundancy: see https://github.com/rdswift/picard/pull/12
Also we don't need `^` / `$` if we use `re.fullmatch()`